api/api_full.go

```diff
@@ -543,6 +543,11 @@ type FullNode interface {
 	// Messages in the `apply` parameter must have the correct nonces, and gas
 	// values set.
 	StateCompute(context.Context, abi.ChainEpoch, []*types.Message, types.TipSetKey) (*ComputeStateOutput, error) //perm:read
+
+	// StateRecomputeTipset recomputes the state of the given tipset, without trying to lookup a pre-computed result
+	// in the chainstore.
+	StateRecomputeTipset(ctx context.Context, tsk types.TipSetKey) (cid.Cid, error) //perm:read
```

Comment:
Is this the only way to achieve this via the rpc? do we not expose enough of the pieces to recompute without adding a new API? We should have a bias _against_ adding new RPC APIs if we can help it.

Reply:
I was also not sure about adding a new API as `StateCompute` was already there, that's why I raised a draft PR to gather everyone's thoughts, I will try to find some other way to do it by using existing APIs.